src/utils/manifest-loader.ts

```diff
@@ -0,0 +1,108 @@
+import path from "node:path";
+import fs from "node:fs";
+import { Log } from "./logger";
+import { TigrisSchema } from "../types";
+import { TigrisFileNotFoundError } from "../error";
+
+const COLL_FILE_SUFFIX = ".ts";
+
+type CollectionManifest = {
+	collectionName: string;
+	schemaName: string;
+	schema: TigrisSchema<unknown>;
+};
+
+type DatabaseManifest = {
+	dbName: string;
+	collections: Array<CollectionManifest>;
+};
+
+/**
+ * Array of databases and collections in each database
+ */
+export type TigrisManifest = Array<DatabaseManifest>;
+
+/**
+ * Loads the databases and schema definitions from file system that can be used
+ * to create databases and collections
+ *
+ * @return TigrisManifest
+ */
+export function loadTigrisManifest(schemasPath: string): TigrisManifest {
+	Log.event(`Scanning ${schemasPath} for Tigris schema definitions`);
+
+	if (!fs.existsSync(schemasPath)) {
+		Log.error(`Invalid path for Tigris schema: ${schemasPath}`);
+		throw new TigrisFileNotFoundError(
+			`Directory not found: ${schemasPath}. Provide a complete path.`
+		);
+	}
+
+	const tigrisFileManifest: TigrisManifest = new Array<DatabaseManifest>();
+
+	// load manifest from file structure
+	for (const schemaPathEntry of fs.readdirSync(schemasPath)) {
+		const dbDirPath = path.join(schemasPath, schemaPathEntry);
+		if (fs.lstatSync(dbDirPath).isDirectory()) {
+			Log.info(`Found DB definition ${schemaPathEntry}`);
+			const dbManifest: DatabaseManifest = {
+				dbName: schemaPathEntry,
+				collections: new Array<CollectionManifest>(),
+			};
+
+			for (const dbPathEntry of fs.readdirSync(dbDirPath)) {
+				if (dbPathEntry.endsWith(COLL_FILE_SUFFIX)) {
+					const collFilePath = path.join(dbDirPath, dbPathEntry);
+					if (fs.lstatSync(collFilePath).isFile()) {
+						Log.info(`Found Schema file ${dbPathEntry} in ${schemaPathEntry}`);
+						const collName = dbPathEntry.slice(
+							0,
+							Math.max(0, dbPathEntry.length - COLL_FILE_SUFFIX.length)
+						);
+						// eslint-disable-next-line @typescript-eslint/no-var-requires,unicorn/prefer-module
+						const schemaFile = require(collFilePath);
+						for (const [key, value] of Object.entries(schemaFile)) {
+							if (canBeSchema(value)) {
+								dbManifest.collections.push({
```

Comment:
it will be tricky to detect but it will be required to distinguish collection/topic here.